### PR TITLE
Added a null value exception in LibvirtBackupSnapshotCommandWrapper.java

### PR DESCRIPTION
**Pull request by manuiiit**

### CODE REVIEW

**Comment by wilderrodrigues** [comment]
You added code to a class that was recently refactored and was covered by unit tests. Please add a new unit test to cover the new flow.

Cheers,
Wilder

Sent from my iPhone

On 22 Jul 2015, at 22:34, manuiiit <notifications@github.com<mailto:notifications@github.com>> wrote:

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/apache/cloudstack/pull/617

Commit Summary

  *   Added a null value exception

File Changes

  *   M plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtBackupSnapshotCommandWrapper.java<https://github.com/apache/cloudstack/pull/617/files#diff-0> (9)

Patch Links:

  *   https://github.com/apache/cloudstack/pull/617.patch
  *   https://github.com/apache/cloudstack/pull/617.diff

File: plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtBackupSnapshotCommandWrapper.java [hunk 1]
```diff
@@ -25,6 +25,8 @@
 import java.io.IOException;
 import java.text.MessageFormat;
 
+import com.thoughtworks.xstream.mapper.Mapper;
+import org.apache.commons.lang.ObjectUtils;
```
Comment (karuturi) [inline plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtBackupSnapshotCommandWrapper.java]: remove the unused imports and format the code as per coding conventions.

**Comment by wilderrodrigues** [comment]
Hi @manuiiit 

I checked the LibvirtBackupSnapshotCommandWrapper class and its coverage, which is still poor compared to the other classes in the KVM hypervisor plugin.

Please, proceed with the changes @karuturi suggested and I will increase the tests on the class in a separate PR.

It is currently 6.9% covered, which is very bad.

Cheers,
Wilder

**Comment by manuiiit** [comment]
Hi Wilder,
          The LibvirtBackupSnapshotCommandWrapper class has no unit
tests.Can I add test cases that covers the exception.

Regards,
P.Maneesha.


On Thu, Jul 23, 2015 at 11:35 AM, Wilder Rodrigues <notifications@github.com

**Comment by wilderrodrigues** [comment]
Hi Maneesha,

Yep, it has... I wrote them.

All the wrappers classes have in total 145 unit tests. Have a look at LibvirtComputingResourceTes, i think that's the name of the class.

As I said, due to complexity of the code that one test is poor, but most of the wrapper classes have 100% coverage.

No worries, I'm busy with improving a couple of tests and will also do the one for the backup snapshot.

Apply the change Rajani suggested so we can get it merged.

Cheers,
Wilder

Sent from my iPhone

On 23 Jul 2015, at 08:55, Maneesha P <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi Wilder,
The LibvirtBackupSnapshotCommandWrapper class has no unit
tests.Can I add test cases that covers the exception.

Regards,
P.Maneesha.


On Thu, Jul 23, 2015 at 11:35 AM, Wilder Rodrigues <notifications@github.com<mailto:notifications@github.com>

**Comment by karuturi** [comment]
merging to master now

**Pull request closed without merging**